Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -160,6 +161,169 @@ public void DispatchValidationFailures_WithRowEditModeControlHavingInvalidValida
 
       validationResultMock.Verify();
+
+      // Validates that dispatch was called to dispatching validator
+      var bocTextValue = ((Control)editableRow).FindControl("0") as BusinessObjectBoundEditableWebControl;
+      validationResultMock.Verify(_ => _.GetValidationFailures(bocTextValue.DataSource.BusinessObject, bocTextValue.Property, true));
+    }
+
+    [Test]
+    public void DispatchValidationFailures_WithListEditModeControlHavingInvalidValidator_HasFailureFromValidatorInRepositoryWithMetadata ()
+    {
+      var values = new[]
+                   {
+                     TypeWithString.Create("\u0001", "A")
+                   };
+
+      var (bocList, simpleColumnDefinition) = InitiateBocListForEditModeWithStringClass(values);
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ControlToValidate = "someBocList";
+      bocList.ID = "someBocList";
+      bocList.SwitchListIntoEditMode();
+
+      Assert.That(bocList.IsListEditModeActive, Is.True);
+
+      var editableRow = ((IBocList)bocList).EditModeController.GetEditableRow(0);
+      ((EditableRow)editableRow).EnsureValidatorsRestored();
+
+      Assert.That(editableRow.HasValidators(), Is.True);
+
+      Assert.That(editableRow.HasValidators(0), Is.True);
+      ControlCollection editableRowValidators = editableRow.GetValidators(0);
+      Assert.That(
+          editableRowValidators.Cast<BaseValidator>().Select(v => v.GetType()),
+          Is.EqualTo(
+              new[]
+              {
+                typeof(ControlCharactersCharactersValidator),
+                typeof(BusinessObjectBoundEditableWebControlValidationResultDispatchingValidator)
+              }));
+
+      ((BaseValidator)editableRowValidators[0]).Validate();
+
+      NamingContainer.Controls.Add(bocList);
+      NamingContainer.Controls.Add(validator);
+
+      var validationResultMock = new Mock<IBusinessObjectValidationResult>();
+      validationResultMock
+          .Setup(e => e.GetValidationFailures(bocList.DataSource.BusinessObject, bocList.Property, true))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>())
+          .Verifiable();
+
+      validationResultMock
+          .Setup(_ => _.GetValidationFailures((IBusinessObject)values[0], simpleColumnDefinition.GetPropertyPath().Properties.Single(), true))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>())
+          .Verifiable();
+      validationResultMock
+          .Setup(_ => _.GetUnhandledValidationFailures(It.IsAny<IBusinessObject>(), It.IsAny<bool>(), It.IsAny<bool>()))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>());
+
+      validator.DispatchValidationFailures(validationResultMock.Object);
+
+      var bocListValidationFailureRepository = bocList.ValidationFailureRepository;
+      var editedRow = ((IBocList)bocList).EditModeController.GetEditableRow(0);
```

Comment:
Is the assert with the `editedRow` (that is now `values[0]`) no longer valid? if so we can remove this line as well